bin/database.py

```diff
@@ -79,6 +79,55 @@ def update_file_origins(cont_list, cont_name):
     update_file_origins(config.db.sessions.find(query), 'sessions')
     update_file_origins(config.db.acquisitions.find(query), 'acquisitions')
 
+def upgrade_to_3():
+    """
+    scitran/core issue #189 - Data Model v2
+
+    Field `metadata` renamed to `info`
+    Field `file.instrument` renamed to `file.modality`
+    Acquisition fields `instrument` and `measurement` removed 
+    """
+
+    def dm_v2_updates(cont_list, cont_name):
+        for container in cont_list:
+            
+            query = {'_id': container['_id']}
+            update = {'$rename': {'metadata': 'info'}}
+
+            files = container.get('files', None)
+            if files is not None:
+                updated_files = []
+                for file in files:
+                    if file.get('metadata', None) is not None:
+                        file['info'] = file.pop('metadata')
+                    if file.get('instrument', None) is not None:
+                        file['modality'] = file.pop('instrument')
+                    updated_files.append(file)
+                update['$set'] = {'files': updated_files}
+
+            if cont_name == 'sessions' and container.get('subject', None) is not None:
+                subject = container['subject']
+                if subject.get('metadata', None) is not None:
+                    subject['info'] = subject.pop('metadata')
+                update['$set'].update({'subject': subject})
+
+            if cont_name == 'acquisitions':
+                update['$unset'] = {'instrument': '', 'measurements': ''}
+
+            result = config.db[cont_name].update_one(query, update)
+
+    query = {'$or':[{'files': { '$exists': True}}, 
+                    {'subject': { '$exists': True}}, 
+                    {'metadata': { '$exists': True}}]}
+
+    dm_v2_updates(config.db.collections.find(query), 'collections')
+    dm_v2_updates(config.db.projects.find(query), 'projects')
+    dm_v2_updates(config.db.sessions.find(query), 'sessions')
+
+    query['$or'].append({'instrument': { '$exists': True}})
+    query['$or'].append({'measurement': { '$exists': True}})
+    dm_v2_updates(config.db.acquisitions.find(query), 'acquisitions')
+
```

Comment:
For performance it could be beneficial to check [Mongodb Aggregation framework](https://docs.mongodb.org/manual/aggregation/)